README.markdown

```diff
@@ -157,6 +157,19 @@ vcsrepo { '/path/to/repo':
 }
 ~~~
 
+To keep local changes while changing revision, use the `keep_local_changes`:
+
+~~~ puppet
+vcsrepo { '/path/to/repo':
+  ensure             => present,
+  provider           => git,
+  source             => 'git://example.com/repo.git',
+  revision           => '0c466b8a5a45f6cd7de82c08df2fb4ce1e920a31',
+  keep_local_changes => true,
+  user               => 'someUser',
+}
+~~~
+
```

Comment:
Since the module is stringified Reference.md will be generated automatically with the required information.
`##### `keep_local_changes`

Valid values: `true`, `false`

Keep local changes on files tracked by the repository when changing revision

Default value: `false``

We can use this command to generate the contents in REFERENCE.md
bundle exec puppet strings generate --format markdown --out REFERENCE.md